includes/woopay/class-woopay-store-api-token.php

```diff
@@ -57,5 +58,25 @@ public function get_args() {
 		public function get_cart_token() {
 			return parent::get_cart_token();
 		}
+
+		//phpcs:disable Generic.CodeAnalysis.UselessOverridingMethod
+		/**
+		 * This function is used to create a custom cart token for verified WooPay users.
+		 *
+		 * @param string $id The user ID.
+		 *
+		 * @return string The cart token.
+		 * @psalm-suppress UndefinedMethod
+		 */
+		public function get_store_api_token_for_user_id( $id ) {
+			return JsonWebToken::create(
+				[
+					'user_id' => $id,
```

Comment:
> Also the name of supported_extensions can become a bit confusing in the future

On https://github.com/Automattic/woocommerce-payments/pull/6562 it's renamed to "adapted extensions", I've been using this name on the WooPay PRs.

> Would we be open to modifying the extensions themselves to, instead of expecting an authenticated user, return the gift cards/points for the user with the validated email?

The extensions themselves need the user to be authenticated to make sure the discount is being applied to the right user, by validated e-mail you mean a flag sent by WooPay that makes the extension identify if the e-mail is verified? I believe we still would need another token for it as it cannot be a single flag because the extension data is a private data (ex anyone could see someone points balance and apply it with this flag).

I don't believe there will be side effects with this PR approach as we authenticate the user as expected for the extension.

Reply:
How long does the token live? Do we have to update or renew the token in WooPay? 

For the regular Cart-Token we update it on every request.

Reply:
The token was without expiration, good catch, on https://github.com/Automattic/woocommerce-payments/pull/6559/commits/fa8d6dc2846dcfddbbcc556594fbeb6be27497eb we add a condition to check the token expiration, it's valid for two weeks [as the cart token](https://github.com/woocommerce/woocommerce-blocks/blob/1b4985ad9a86b083d7f6fa6e6bd5dde432203478/src/StoreApi/Routes/V1/AbstractCartRoute.php#L211).

Reply:
Updated the PR to use a WooPay email flag instead of another cart token on https://github.com/Automattic/woocommerce-payments/pull/6559/commits/ce78575f582990cde6d788be348d2834ce29d920.